src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1735,10 +1813,13 @@ SubCmdValidate::execute(const SubCmdOptions& _options) const
 
     // Logic for individually defined tests
     // Enqueue the matching test suites to be executed
-    std::string testSuiteName = get_test_name(testSuite[index].ptTest.get("name","<unknown>"));
     for (const auto & testName : testsToRun) {
       if (testName.compare(testSuiteName) == 0) {
         testObjectsToRun.push_back(&testSuite[index]);
+        // add custom param to the ptree if available
+        if (!param.empty() && testName.compare(param[0]) == 0) {
```

Comment:
```suggestion
        if (!param.empty() && boost::equals(testName, param[0])) {
```
Same thing here I think this looks nicer!